src/NodaTime/DateInterval.cs

```diff
@@ -87,6 +92,73 @@ public DateInterval(LocalDate start, LocalDate end)
         }
 
         /// <summary>
+        /// Returns the hash code for this interval, consistent with <see cref="Equals(DateInterval)"/>.
+        /// </summary>
+        /// <returns>The hash code for this interval.</returns>
+        public override int GetHashCode() =>
+            HashCodeHelper.Initialize()
+                .Hash(Start)
+                .Hash(End)
+                .Hash(Inclusive)
+                .Value;
+
+        /// <summary>
+        /// Compares two <see cref="DateInterval" /> values for equality.
+        /// </summary>
+        /// <remarks>
+        /// Date intervals are equal if they have the same start and end dates and are both inclusive or both exclusive:
+        /// an exclusive date interval of [2001-01-01, 2001-02-01) is not equal to the inclusive date interval of
+        /// [2001-01-01, 2001-01-31], even though both contain the same range of dates.
+        /// </remarks>
+        /// <param name="lhs">The first value to compare</param>
+        /// <param name="rhs">The second value to compare</param>
+        /// <returns>True if the two date intervals have the same properties; false otherwise.</returns>
+        public static bool operator ==(DateInterval lhs, DateInterval rhs)
+        {
+            if (ReferenceEquals(lhs, rhs))
+            {
+                return true;
+            }
+            if (ReferenceEquals(lhs, null) || ReferenceEquals(rhs, null))
```

Comment:
I assume that this is correct, but I note that we don't implement `operator==` for any other classes, so I don't have anything to check against.